server/controller/cycleData/persistNodeValues.ts

```diff
@@ -0,0 +1,38 @@
+import { CountryIso } from '@meta/area'
+import { Assessment, Cycle } from '@meta/assessment'
+import { TableData } from '@meta/data'
+import { User } from '@meta/user'
+
+import { persistNodeValue } from './persistNodeValue'
+
+interface Props {
+  assessment: Assessment
+  cycle: Cycle
+  user: User
+  tableData: TableData
+  tableName: string
+}
+
+// Wrapper to support persisting full tables
+export const persistNodeValues = async (props: Props): Promise<void> => {
+  const { assessment, cycle, user, tableData, tableName } = props
+  const promises = Object.entries(tableData).flatMap(([countryIso, countryValues]) =>
+    Object.entries(countryValues).flatMap(([_section, sectionValues]) =>
+      Object.entries(sectionValues).flatMap(([colName, colValues]) =>
+        Object.entries(colValues).flatMap(([variableName, value]) => {
+          return persistNodeValue({
```

Comment:
I'm wondering if there's a way to remove all this iterations.
what if the estimation engine returns an array of nodeUpdates e.g.
 ```typescript
[
{countryIso, colName, tableName, assessment, cycle, variableName, value},
...
]
```
 ?
 This way, the returned array is input to this persistNodeValues method and Props would become:
 ```typescript
 interface Props {
   nodes: Array<NodeUpdate>
   user: User
 }
 ```